Hazel/Udp/UnityUdpClientConnection.cs

```diff
@@ -85,19 +85,20 @@ protected override void WriteBytesToConnection(byte[] bytes, int length)
             }
         }
 
-        private void WriteBytesToConnectionReal(byte[] bytes, int length)
+        private void WriteBytesToConnectionReal(SmartBuffer bytes, int length)
         {
             try
             {
+                bytes.AddUsage();
                 this.Statistics.LogPacketSend(length);
                 socket.BeginSendTo(
-                    bytes,
+                    (byte[])bytes,
                     0,
                     length,
                     SocketFlags.None,
                     EndPoint,
                     HandleSendTo,
-                    null);
+                    bytes);
             }
             catch (NullReferenceException) { }
```

Comment:
If this throws and the AsyncCallback doesn't fire, is the Connection always just disposed? If not we may need to make sure this SmartBuffer gets recycled in the catch blocks...

Reply:
Most likely yeah, but good to throw the recycles in there anyway.